sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -716,22 +717,38 @@ ProgramManager::collectDependentDeviceImagesForVirtualFunctions(
       // virtual-functions-set properties, but their handling is the same: we
       // just grab all sets they reference and add them for consideration if
       // we haven't done so already.
+      bool isDummyImage = false;
       for (const sycl_device_binary_property &VFProp :
```

Comment:
So, we started with a device image which contains a kernel, obtained sets of virtual functions that it uses and now we iterate over device images from those sets, right?

We expect to see a device image with the `dummy` property, but your other change below to `ProgramManager::addImages` seems to make sure that no `dummy` images are ever added into `set2binimage` map. Am I missing something here, or there is a bug?